src/main/cpp/libclp_ffi_java/ir_stream/common.tpp

```diff
@@ -43,20 +31,9 @@ namespace libclp_ffi_java::ir_stream {
             jint time_zone_id_length,
             jlong reference_timestamp
     ) {
-        if (timestamp_pattern_length < 0 || timestamp_pattern_syntax_length < 0 ||
```

Comment:
At least it is not very clear to me why we don't need to do input validation anymore. isn't it called from JNIEXPORT jbyteArray JNICALL
Java_com_yscope_clp_irstream_FourByteClpIrOutputStream_encodePreambleNative?

Reply:
This is a bit of a borderline-case. At first, this code was in `Java_com_yscope_clp_irstream_FourByteClpIrOutputStream_encodePreambleNative` which is directly called by the JVM. The caller of this method would be using `timestampPattern.length`, so there should be little chance of an error that would require validation.

However, this method was redundant between the four-byte and eight-byte encodings, so I extracted it into a method. Now that it's a separate method, I think you're alluding that it may not be called directly by the JVM, so I'll put the validation back to be safe in this case.

For other cases where the method is called directly by the JVM, I still think we should omit this type of validation.

Reply:
or you can add a comment to explain that JVM will ensure the argument is valid. it just wasn't obvious to me

If the code fails loudly, I think it is totally fine to trust programmer and remove the check,

Reply:
I think I'll leave it in this case, for now. If we see a performance penalty later, we can remove it. Don't want to prematurely optimize, lol.